### DISCUSSION TITLE
Feature Request: Postgres: Add possibility to run "low level" queries directly via psql driver

### DISCUSSION THREAD

**Discussion by dberardo-com** [duckdb/duckdb#6223] [Ideas]
Related to: https://github.com/duckdb/duckdb/discussions/6108

---

Use case:

- i need to perform some action when a Large table changes in postgres
- i could do it via NOTIFY/LISTEN or by enabling commit_timestamps ( https://dba.stackexchange.com/questions/58214/getting-last-modification-date-of-a-postgresql-database-table)
- i went for the second alternative given that i dont think duckdb implements NOTIFY/LISTEN functions
- i run SELECT pg_xact_commit_timestamp(xmin) FROM  mytable and get this error:
CatalogException: Catalog Error: Scalar Function with name pg_xact_commit_timestamp does not exist!

The same query works fine on the DB

Questions:

is it possible to tell duckdb to run a query directly via the db driver (so that the function pg_xact_commit_timestamp is run on the db directly?
does duckdb implement listen/notify, and if so, do they also work on the underlying storage engine (here postgres)?

another use case would be to run queries using underlying extensions, like postGIS or TimescaleDB via duckdb

**Comment by Mytherin** [comment]
Thanks for the feature request! I think this would be a good feature and we have discussed implementing this ourselves before. Perhaps you could file it in the [postgres scanner repository](https://github.com/duckdblabs/postgres_scanner), however?

**Reply by dberardo-com** [reply]
thanks, i will be post that there and link to this discussion. sorry for publishing a reduntant issue here, but i thought you might have not gotten any notification from github when i first opened this discussion. that issue could be closed IMO.

**Reply by dberardo-com** [reply]
unfortunately no answer there yet.

btw, i have noticed that the postgres_attach query takes very long time to execute (1+ minute) is there a way to speed up the connection process ?

**Comment by dberardo-com** [comment]
https://github.com/duckdblabs/postgres_scanner/issues/78
